### PR TITLE
Update bytecode verification to support OpenJDK VarHandles

### PR DESCRIPTION
**Pull request by babsingh**
For OpenJDK VarHandles, the invokevirtual calls to the VarHandle
polymorphic methods are rewritten to the invokehandle bytecode in
ClassFileOracle (#10557). Bytecode verification has been updated to
handle this conversion. These changes are identical to the existing
MethodHandle handling.

Related: #7352

Co-authored-by: Jack Lu <Jack.S.Lu@ibm.com>
Signed-off-by: Babneet Singh <sbabneet@ca.ibm.com>

### CODE REVIEW

**Comment by babsingh** [comment]
@DanHeidinga can you please review this pull request?
fyi @fengxue-IS @tajila

File: runtime/bcverify/rtverify.c [hunk 1]
```diff
@@ -1531,15 +1531,26 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
...
-				type = POP;		/* Remove the receiver from the stack */
-				rc = isClassCompatibleByName (verifyData, type, (U_8 *)"java/lang/invoke/MethodHandle", sizeof("java/lang/invoke/MethodHandle") - 1, &reasonCode);
+				rc = isClassCompatibleByName(verifyData, type, (U_8 *)"java/lang/invoke/MethodHandle", sizeof("java/lang/invoke/MethodHandle") - 1, &reasonCode);
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+				if (FALSE == rc) {
+					if (BCV_ERR_INSUFFICIENT_MEMORY == reasonCode) {
+						goto _outOfMemoryError;
+					}
+					/* Receiver compatible with VarHandle? */
+					rc = isClassCompatibleByName(verifyData, type, (U_8 *)"java/lang/invoke/VarHandle", sizeof("java/lang/invoke/VarHandle") - 1, &reasonCode);
```
Comment (DanHeidinga) [inline runtime/bcverify/rtverify.c:1544]: is this applicable to both the `JBinvokehandle` and the  `JBinvokehandlegeneric` cases?

Comment (babsingh) [reply runtime/bcverify/rtverify.c:1544]: Just the `invokehandle` scenario. I should skip the `VarHandle` check for `JBinvokehandlegeneric`.

File: runtime/bcverify/rtverify.c [hunk 2]
```diff
@@ -1531,15 +1531,26 @@ verifyBytecodes (J9BytecodeVerificationData * verifyData)
...
 				 */
 
+				/* Remove the receiver from the stack. */
+				type = POP;
 				/* Receiver compatible with MethodHandle? */
-				type = POP;		/* Remove the receiver from the stack */
-				rc = isClassCompatibleByName (verifyData, type, (U_8 *)"java/lang/invoke/MethodHandle", sizeof("java/lang/invoke/MethodHandle") - 1, &reasonCode);
+				rc = isClassCompatibleByName(verifyData, type, (U_8 *)"java/lang/invoke/MethodHandle", sizeof("java/lang/invoke/MethodHandle") - 1, &reasonCode);
+#if defined(J9VM_OPT_OPENJDK_METHODHANDLE)
+				if ((JBinvokehandle == bc) && (FALSE == rc)) {
```
Comment (babsingh) [inline runtime/bcverify/rtverify.c:1539]: ^^^ Updated to perform the `VarHandle` check only for `JBinvokehandle`.

**Review by DanHeidinga — APPROVED** [review]
lgtm

**Comment by DanHeidinga** [comment]
Jenkins test sanity plinux jdk8,jdk11

**Pull request merged**